src/WebJobs.Script.WebHost/Diagnostics/AzureMonitorDiagnosticLogger.cs

```diff
@@ -42,7 +44,7 @@ public AzureMonitorDiagnosticLogger(string category, string hostInstanceId, IEve
         public bool IsEnabled(LogLevel logLevel)
         {
             // We want to instantiate this Logger in placeholder mode to warm it up, but do not want to log anything.
-            return !string.IsNullOrEmpty(_hostNameProvider.Value) && !_environment.IsPlaceholderModeEnabled();
+            return _options.Value.AzureMonitorEnabled && !string.IsNullOrEmpty(_hostNameProvider.Value) && !_environment.IsPlaceholderModeEnabled();
```

Comment:
One area of focus in my benchmarking is comparisons of a vanilla ASP.NET hello world app and a similar Function App. To do so, we need the ability to disable all the optional features, like this, auth, app insights, etc. to get an apples to apples comparison. Also I think its good in general to allow configurability of non-essential components, so we have a "pay to play" model for performance sensitive scenarios.

Reply:
Removed the flag